plugins/broadcast/server/server.go

```diff
@@ -0,0 +1,164 @@
+package server
+
+import (
+	"github.com/cockroachdb/errors"
+	"github.com/iotaledger/hive.go/node"
+	"sync"
+
+	"github.com/iotaledger/goshimmer/packages/tangle"
+	"go.uber.org/atomic"
+
+	"github.com/iotaledger/hive.go/events"
+	"github.com/iotaledger/hive.go/netutil/buffconn"
+
+	"fmt"
+	"io"
+	"net"
+)
+
+type connection struct {
+	bufferedConn *buffconn.BufferedConnection
+	log          *node.Plugin //Is just used for logging
+	active       bool
+}
+
+var (
+	connectionList      = [256]connection{}
+	connectionListMutex sync.RWMutex
+	index               atomic.Uint32
+)
+
+// Listen starts a TCP listener and starts a Connection for each accepted connection
+func Listen(bindAddress string, log *node.Plugin, shutdownSignal <-chan struct{}) error {
+	listener, err := net.Listen("tcp", bindAddress)
+	if err != nil {
+		return fmt.Errorf("failed to start Broadcast daemon: %w", err)
+	}
+
+	go func() {
+		for {
+			connection, err := listener.Accept()
+			if err != nil {
+				if connection != nil {
+					log.LogInfof("Couldn't accept connection: %s", err)
+				}
```

Comment:
`connection` will probably ending up `nil` if `err` is `nil`, so remove this if check on the `connection`